modules/internal/ChapelArray.chpl

```diff
@@ -2037,7 +2061,18 @@ module ChapelArray {
             hi = this.domain.high+1;
       const newDom = {lo..hi};
       on this._value {
-        this._value.dsiReallocate(newDom);
```

Comment:
'dsiReallocate' takes a domain as an argument, so either 'newDom' needs to be a domain, or one needs to be created from it.